### ISSUE TITLE
fail to encode array of nested classes

### ISSUE THREAD

**Issue by FloorLamp** [dignifiedquire/borc#27]
I have 2 classes with custom encoding:

```
class Serializable {
  serialize () {
    const encoder = new cbor.Encoder()
    this.encodeCBOR(encoder)
    return encoder.finalize()
  }

  static deserialize (serialized) {
    return cbor.decodeAll(serialized)
  }
}

class TestA extends Serializable {
  encodeCBOR (gen) {
    return gen.write(44)
  }
}

class TestB extends Serializable {
  encodeCBOR (gen) {
    return gen.write([
      new TestA()
    ])
  }
}
```

This works fine and `TestB` serializes to `<Buffer 81 18 2c>`. If I update `TestB` to this:

```
class TestB extends Serializable {
  encodeCBOR (gen) {
    return gen.write([
      new TestA(),
      true,
    ])
  }
}
```

Then it serializes to `<Buffer 82 18 2c>`, which has the correct length but not the new element. Attempting to decode that throws a `Error: Undeterminated nesting`.

**Comment by daviddias** [comment]
@dignifiedquire mind taking a look?

**Comment by dignifiedquire** [comment]
Thanks for the report, fixed in #28
